.github/workflows/example-cached.yml

```
@@ -0,0 +1,203 @@
+          # Key includes content hash for automatic invalidation
+          key: promptfoo-${{ runner.os }}-${{ hashFiles('prompts/**', 'configs/**', '**/promptfooconfig.yaml') }}-${{ github.sha }}
+          # Fallback keys for partial cache hits
+          restore-keys: |
+            promptfoo-${{ runner.os }}-${{ hashFiles('prompts/**', 'configs/**', '**/promptfooconfig.yaml') }}-
+            promptfoo-${{ runner.os }}-
+
+      # Step 3: Cache API responses with weekly rotation
+      # This provides a balance between cache freshness and API cost savings
+      - name: Get cache rotation key
+        id: cache-key
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2129:style:1:1: Consider using { cmd1; cmd2; } >> file instead of individual redirects

(shellcheck)